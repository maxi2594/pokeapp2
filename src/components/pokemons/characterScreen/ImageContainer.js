import React, { useState } from 'react'
import PropTypes from "prop-types";

export const ImageContainer = ({ data }) => {

    const [imageSrc, setImageSrc] = useState(data.sprites.front_default)


    const sprites = (Object.keys(data.sprites).map(key => ({
        sprite: [key],
        src: data.sprites[key]
    })))

    return (
        <div className='imageContainer'>
            <img
                style={{ width: 250 }}
                src={imageSrc}
                alt={data.pokemon.name}
            />
            <div className="imageSelector">
                <ul>
                    {
                        sprites.map(elem => (
                            elem.src ?
                                (<button
                                    key={elem.sprite}
                                    onClick={() => setImageSrc(elem.src)}
                                >{elem.sprite}</button>)
                                :
                                null
                        )
                        )
                    }
                </ul>
            </div>
        </div>
    )
}

ImageContainer.propTypes = {
    data: PropTypes.object.isRequired
}
