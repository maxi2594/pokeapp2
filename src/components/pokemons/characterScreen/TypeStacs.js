import React, { useEffect, useState } from 'react'
import { getPokes } from '../../../helpers/getPokesApi'

export const TypeStacs = ({ src }) => {

    const [data, setdata] = useState({
        data: null,
        error: null
    })

    // useEffect(() => {
    //     getPokes(src).then(data => data.damage_relations.double_damage_from.map(elem=> console.log(elem)))
    // }, [src]);

    return (
        data ?
            (<div className='typeStacs'>
                <div className='stats'>
                    <h3 style={{ width: '100%' }}>Damage Relations</h3>
                    <label> Doble Damage From: </label>
                    {/* {
                    data.damage_relations.double_damage_from.map( damage => (
                     <span>{damage.name}</span>       
                    ))
                } */}
                </div>
            </div>)
            :
            null
    )
}
