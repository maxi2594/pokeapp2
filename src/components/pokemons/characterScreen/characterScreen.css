.flexContainer {
  display: flex;
  flex-flow: wrap;
  width: 100%;
  height: 100%;
  background-color: transparent;
}
.imageContainer {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 40%;
  height: 100%;
  padding: 2rem;
}
.imageSelector {
  align-items: center;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}
button {
  border-radius: 1rem;
  background-color: transparent;
  border: 2px solid grey;
  color: grey;
}
