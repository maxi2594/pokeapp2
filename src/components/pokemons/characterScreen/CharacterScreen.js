import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router'

import { getPokes } from '../../../helpers/getPokesApi'
import { ImageContainer } from './ImageContainer';
import { TypeStacs } from './TypeStacs';
import "./characterScreen.css";

export const CharacterScreen = () => {

    const [data, setData] = useState(null)
    const [typeStacs, settypeStacs] = useState({
        show: true,
        src: ''
    })

    const { id } = useParams()

    const handeSetTypeStac = (src) => {
        settypeStacs(s => ({
            ...s,
            src
        }))
    }

    useEffect(() => {

        const data = getPokes(`https://pokeapi.co/api/v2/pokemon-form/${id}`)
        data.then(data => {
            setData(data)
        })

    }, [id])

    return (
        data
        &&
        (<div className='flexContainer'>

            <ImageContainer data={data} />

            <div className='statsContainer'>
                <div className='stats'>
                    <label> name : </label>
                    <p data-testid="name">{data.name.toUpperCase()}</p>
                </div>

                <div className='stats'>
                    <label>Battle Only :</label>
                    <p> {data.is_battle_only ? 'SI' : 'NO'}</p>
                </div>

                <div className='stats'>
                    <label>Types:</label>
                    <div>
                        {
                            data.types.map(elem =>
                            (<button
                                key={elem.slot}
                                onClick={() => handeSetTypeStac(elem.type.url)}
                            >{elem.type.name}</button>)
                            )
                        }
                    </div>
                </div>
                {
                    typeStacs.show && (<TypeStacs src={data.types[0].type.url} />)
                }

            </div>
        </div>

        )
    )
}
