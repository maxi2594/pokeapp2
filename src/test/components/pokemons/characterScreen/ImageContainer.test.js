import { shallow } from "enzyme"
import { MemoryRouter } from "react-router"
import { ImageContainer } from "../../../../components/pokemons/characterScreen/ImageContainer"
import { data } from "../../../fakeData/fakeData"


describe('test inj <ImageContainer/>', () => {

    const wrapper = shallow(
        <MemoryRouter initialEntries={[{ location: { key: 'oa23iu' } }]}>
            <ImageContainer data={data} />
        </MemoryRouter>
    )

    test('should show properly', () => {

        expect(wrapper).toMatchSnapshot()
        console.log(wrapper.find('div').length);
    })

})
