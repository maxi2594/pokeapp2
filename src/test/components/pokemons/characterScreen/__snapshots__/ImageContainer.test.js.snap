// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test inj <ImageContainer/> should show properly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "dze7bj",
          "location": Object {
            "key": "oa23iu",
          },
          "pathname": "/",
          "search": "",
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "dze7bj",
        "location": Object {
          "key": "oa23iu",
        },
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ImageContainer
    data={
      Object {
        "damage_relations": Object {
          "double_damage_from": "somePokeDamage",
        },
        "is_battle_only": true,
        "name": "somepokename",
        "pokemon": Object {
          "name": "somepokename",
        },
        "sprites": Object {
          "front_default": "http://fakeUrl/image",
        },
        "types": Array [
          Object {
            "slot": 1,
            "type": Object {
              "name": "name",
              "url": "https:/faakeUrl.com",
            },
          },
          Object {
            "slot": 2,
            "type": Object {
              "name": "name",
              "url": "https:/faakeUrl.com",
            },
          },
          Object {
            "slot": 3,
            "type": Object {
              "name": "name",
              "url": "https:/faakeUrl.com",
            },
          },
        ],
      }
    }
  />
</Router>
`;
